### PR TITLE
Adding a use cases document

### LINKED ISSUE #81: Document use cases
We need a document that outlines use cases for the HTTP add on.

### Use-Case

The document should be used by users (prospective and current) of the HTTP add on to guide their decision to use the project, and by developers to decide if/how a feature will fit.

### Specification

- [ ] A new document with use cases on it
- [ ] A link from the README to this new document

### PR DESCRIPTION
**Pull request by arschles**
This PR adds a document describing use cases for the HTTP add on.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #81

### CODE REVIEW

**Comment by arschles** [comment]
@tomkerkhove we discussed this document in the March 2, 2021 developer standup. I've submitted this PR as a draft. Can you please take a look and let me know if this is the kind of thing you had in mind?

File: docs/use_cases.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Common Use Cases
+
+This document includes several common scenarios in which this project may be deployed along with descriptions on why and how it could be deployed in each case.
+
+## Current Containerized HTTP Application In The Cloud, Migrating to Kubernetes
+
+In this use case, an application may be containerized running on a managed cloud platform that supports containers. Below is a non-exhaustive, non-ordered list of some examples:
+
+- [Azure App Services](https://docs.microsoft.com/en-us/azure/app-service/quickstart-custom-container?pivots=container-linux)
```
Comment (tomkerkhove) [inline docs/use_cases.md:9]: Nit - Let's alphabetically sort to not be biased

**Review by tomkerkhove — COMMENTED** [review]
I think it's a good start but would add some schematics of the Kubernetes resources that comes into play and use color-coding who's in charge of what.

Would that make sense?

**Comment by arschles** [comment]
@tomkerkhove I think so. I'll write that for one of these use cases and ask you for another review

**Comment by arschles** [comment]
@tomkerkhove I added some color coding. I'm not exactly sure what you mean by "schematics" of Kubernetes resources though. Do you mean a YAML reference (if so, I added that in https://github.com/kedacore/http-add-on/pull/78/files)

File: docs/use_cases.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+This document includes several common scenarios in which this project may be deployed along with descriptions on why and how it could be deployed in each case.
+
+## Current Containerized HTTP Application In The Cloud, Migrating to Kubernetes
+
+In this use case, an application may be containerized running on a managed cloud platform that supports containers. Below is a non-exhaustive, alphabetically-ordered list of some examples:
+
+- [Amazon ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html)
+- [Azure App Services](https://docs.microsoft.com/en-us/azure/app-service/quickstart-custom-container?pivots=container-linux)
+- - [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform/)
```
Comment (tomkerkhove) [inline docs/use_cases.md:11]:
```suggestion
- [Digital Ocean App Platform](https://www.digitalocean.com/products/app-platform/)
```

**Review by tomkerkhove — APPROVED** [review]
LGTM, the schematics I was referring to was the one you've added with color-coding.

Eventually we should expand the use cases with service mesh, and all once we support it

**Comment by arschles** [comment]
@tomkerkhove wonderful. And, agreed. I added a comment at https://github.com/kedacore/http-add-on/issues/6#issuecomment-801237691 reminding us that we need to document how service meshes integrate with the HTTP add on, when we do implement it.

**Pull request merged**